widgetssdk/src/test/java/com/glia/widgets/core/permissions/PermissionManagerTest.kt

```diff
@@ -0,0 +1,472 @@
+package com.glia.widgets.core.permissions
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager.PERMISSION_DENIED
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.os.Build
+import com.glia.androidsdk.Engagement
+import com.glia.androidsdk.GliaException
+import com.glia.widgets.permissions.Permissions
+import com.glia.widgets.permissions.PermissionsGrantedCallback
+import com.glia.widgets.permissions.PermissionsRequestRepository
+import com.glia.widgets.permissions.PermissionsRequestResult
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+class PermissionManagerTest {
+    private lateinit var context: Context
+    private lateinit var checkSelfPermission: CheckSelfPermission
+    private lateinit var permissionsRequestRepository: PermissionsRequestRepository
+
+    private lateinit var permissionManager: PermissionManager
+
+    @Before
+    fun setUp() {
+        context = mock()
+        checkSelfPermission = mock()
+        permissionsRequestRepository = mock()
+
+        permissionManager = PermissionManager(
+            context,
+            checkSelfPermission,
+            permissionsRequestRepository,
+            Build.VERSION_CODES.R
+        )
+    }
+
+    // .getPermissionsForEngagementMediaType()
```

Comment:
I think this can be removed. 😉

Reply:
Ah, it probably means that below is a group of tests for `getPermissionsForEngagementMediaType()`. Maybe it might be better to add several words like "Tests for ...". Otherwise it looks like disabled code.

Reply:
I really miss the `// MARK:` in Kotlin/Java.
I just wanted to mark different blocks as paragraphs in class.